.github/workflows/e2e_aws.yaml

```diff
@@ -0,0 +1,64 @@
+name: End-to-End AWS Tests
+
+on:
+ workflow_dispatch:
+ push:
+   branches: [main]
+
+env:
+  GOOS: linux
+  GO111MODULE: on
+
+jobs:
+  test-aws:
+    name: Test AWS Cloud Deployment
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_KEY }}
+      AWS_DEFAULT_REGION: "us-east-1"
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check if environment variables are set  # Fails workflow if the secrets are not set
+        run: |
+          if [[ -z "$AWS_ACCESS_KEY_ID" ]]; then
+            echo "AWS_ACCESS_KEY_ID is not set. Please check if secrets.AWS_ACCESS_KEY is in the repository"
```

Comment:
make an emphasis that this is likely caused by branch being out of the upstream repo.